packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md

```diff
@@ -54,63 +54,180 @@ Therefore, when dealing with Custom Roles, these three entities always exist:
-  ```
-
-* [A specific App](/docs/reference/api/apps/#get-application)
-
-  ``` http
-  https://${yourOktaDomain}/api/v1/apps/${targetAppId}
-  ```
+  The service in which the resource belongs to. Each resource only belongs to one service.
+  The following are the supported services.
+
+| Service           |  ORN service value    |
+| ----------------- | --------------------- |
+| Directory         | directory             | 
+| Identity Provider | idp                   |
+
+###### tenantId
+
+Tenant Id will be the tenant that is using the service, which currently is just your [org's id](/docs/reference/api/org/#get-org-settings).
+
+###### objectType
+
+Object type is specific to the service, for example we have object types `groups` and `users` for the `directory` service.
+
+###### objectId
+
+The object's identifier.
+
+###### contained_resources
+
+`contained_resources` is an ORN specific segment which targets all of the resources in the container resource. For example,
+
+`orn:okta:directory:${yourOrgId}:groups:123:contained_resources`
+
+Group 123 is the example container resource, since `:contained_resources` is used we are targeting the users in the group, rather than the group itself.
+
+#### REST URL
+
+If the resource has a corresponding Okta API you can specify a resource by their REST URL as well. Not all resources will
+have a corresponding Okta API, we recommend using ORNs for such resources.
+
+#### Supported Resources
+
+The following are the supported resources.
+
+<table>
+  <tr>
+    <td>Service</td>
+    <td>Resource</td>
+    <td>REST URL Resource Identifier</td>
+  </tr>
+  <tr>
+    <td>Directory</td>
+    <td>All Users</td>
+    <td><a href="/docs/reference/api/users/#list-users">https://${yourOktaDomain}/api/v1/users</a></td>
+  </tr>
+  <tr>
+    <td />
+    <td>All Groups</td>
+    <td><a href="/docs/reference/api/groups/#list-groups">https://${yourOktaDomain}/api/v1/groups</a></td>
+  </tr>
+  <tr>
+    <td />
+    <td>A specific Group</td>
+    <td><a href="/docs/reference/api/groups/#get-group">https://${yourOktaDomain}/api/v1/groups/${groupId}</a></td>
+  </tr>
+  <tr>
+    <td />
+    <td>All Users within a specific Group</td>
+    <td><a href="/docs/reference/api/groups/#list-group-members">https://${yourOktaDomain}/api/v1/groups/${groupId}/users</a></td>
+  </tr>
+  <tr>
+    <td>Identity Provider</td>
+    <td>All Apps</td>
+    <td><a href="/docs/reference/api/apps/#list-applications">https://${yourOktaDomain}/api/v1/apps</a></td>
+  </tr>
+  <tr>
+    <td />
+    <td>All Apps of a specific type</td>
+    <td><a href="/docs/reference/api/apps/#list-apps-by-name">https://${yourOktaDomain}/api/v1/apps/?filter=name+eq+\"${targetAppType}\"</a></td>
+  </tr>
+  <tr>
+    <td />
+    <td>A specific App</td>
+    <td><a href="/docs/reference/api/apps/#get-application">https://${yourOktaDomain}/api/v1/apps/${appId}</a></td>
+  </tr>
+</table>
+
+#### Early Access Resources
+
+<ApiLifecycle access="ea" />
```

Comment:
I'd consider adding both in the same table, that will clearly tell the developer where ORN can and cannot be used. Also it seems we are supporting more resources than the ones listed here? or is the plan to add those in a separate PR?

Reply:
This is piggy backed with the EA-self service permission/resources we're adding. Yeah the other ones they will need to be added when they're ready to actually EA.